src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java

```diff
@@ -0,0 +1,764 @@
+          sweepMin = mapper.getMinX() + 1;
+          sweepMax = mapper.getMaxX() - 1;
+
+          // if pos is too small or too big flip direction
+          if (sweepPos >= sweepMax || sweepPos <= sweepMin) {
+            sweepStep = sweepStep * -1;
+          }
+
+          sweepPos += sweepStep;
+          moveTo(sweepPos);
+          Thread.sleep(sweepDelay);
+        }
+      } catch (Exception e) {
+        isSweeping = false;
+      }
+    }
+
+  }
+
+  public final static Logger log = LoggerFactory.getLogger(AbstractServo.class);
+
+  private static final long serialVersionUID = 1L;
+
+  public static void main(String[] args) throws InterruptedException {
+    try {
+
+      Runtime.start("gui", "SwingGui");
+      Platform.setVirtual(false);
+
+      Arduino mega = (Arduino) Runtime.start("mega", "Arduino");
+      mega.setBoardMega();
+      HobbyServo servo = (HobbyServo) Runtime.start("servo", "HobbyServo");
+      servo.setPin(12);
+      servo.sweepDelay = 3;
+      // servo.save();
+      servo.load();
+      servo.save();
+      log.info("sweepDely {}", servo.sweepDelay);
+
+    } catch (Exception e) {
+      log.error("main threw", e);
+    }
+  }
+
+  /**
+   * acceleration of servo - not really implemented
+   */
+  protected Double acceleration;
+
+  /**
+   * 
+   */
+  protected Boolean autoDisable = false;
+
+  /**
+   * list of controller names this control is attached to
+   */
+  protected Set<String> controllers = new LinkedHashSet<>();
+
+  /**
+   * the current input position (pre-mapper)
+   */
+  protected Double currentPosInput;
+
+  /**
+   * if enabled then a pwm pulse is keeping the servo at the current position,
+   * and movements are possible
+   */
+  protected boolean enabled = false;
+
+  /**
+   * The servos encoder - by "default" this will be a TimerEncoder - where a
+   * timer calculates the expected time the servo will make and complete
+   * movements, and some configured division of time is configured to send
+   * position update events.
+   * 
+   * Unlike previous servo events which dependend on feedback coming from
+   * Arduino, TimerEncoder events do not need an Arduino, in fact they don't
+   * even need anything as computers are pretty good at counting time.
+   * 
+   * Arduino MrlEncoders (legacy) could potentially be substituted for a
+   * TimerEncoder, however there is
+   * 
+   */
+  EncoderControl encoder;
+
+  protected boolean isSweeping = false;
+
+  /**
+   * last time the servo has moved
+   */
+  protected long lastActivityTime = 0;
+
+  /**
+   * servo's last position
+   */
+  protected Double lastPos;
+
+  /**
+   * list of servo listeners names
+   */
+  protected transient Set<String> listeners = new LinkedHashSet<>();
+
+  /**
+   * input mapper
+   */
+  protected Mapper mapper;
+
+  /**
+   * maximum speed
+   */
+  protected Double maxSpeed;
+
+  /**
+   * status if the servo thinks its moving ..
+   */
+  protected boolean moving;
```

Comment:
seems difficult to compute as know as a boolean.. this should be removed as the logic for computing that a servo is moving would require testing if it's sweeping.. or if it's moving to a position with a set velocity..